### PR TITLE
Feat: Add free text search to '--exclude' param

### PR DESCRIPTION
**Pull request by tech-dm-klymenko**
Feature for #8210 

## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-config/src/node/ConfigParser.ts [hunk 1]
```diff
@@ -259,11 +269,11 @@ export default class ConfigParser {
         // when CLI --spec is explicitly specified, this._config.specs contains the filtered
         // specs matching the passed pattern else the specs defined inside the config are returned
         let specs = ConfigParser.getFilePaths(this._config.specs!, this._config.rootDir, this._pathService)
-        let exclude = ConfigParser.getFilePaths(this._config.exclude!, this._config.rootDir, this._pathService)
+        let exclude = allKeywordsContainPath(this._config.exclude!) ? ConfigParser.getFilePaths(this._config.exclude!, this._config.rootDir, this._pathService) : this._config.exclude!
```
Comment (christian-bromann) [inline packages/wdio-config/src/node/ConfigParser.ts:272]:
```suggestion
        let exclude = allKeywordsContainPath(this._config.exclude!)
            ? ConfigParser.getFilePaths(this._config.exclude!, this._config.rootDir, this._pathService)
            : this._config.exclude || []
```

File: packages/wdio-config/src/node/ConfigParser.ts [hunk 2]
```diff
@@ -259,11 +269,11 @@ export default class ConfigParser {
...
         const suites = Array.isArray(this._config.suite) ? this._config.suite : []
 
         // only use capability excludes if (CLI) --exclude or config exclude are not defined
-        if (Array.isArray(capExclude) && exclude.length === 0){
+        if (Array.isArray(capExclude) && exclude!.length === 0){
```
Comment (christian-bromann) [inline packages/wdio-config/src/node/ConfigParser.ts:276]: Added a `|| []` above
```suggestion
        if (Array.isArray(capExclude) && exclude.length === 0){
```

File: packages/wdio-config/src/node/ConfigParser.ts [hunk 3]
```diff
@@ -307,7 +317,7 @@ export default class ConfigParser {
         }
 
         return this.shard(
-            this.filterSpecs(specs, <string[]>exclude)
+            this.filterSpecs(specs, <string[]>exclude!)
```
Comment (christian-bromann) [inline packages/wdio-config/src/node/ConfigParser.ts:320]:
```suggestion
            this.filterSpecs(specs, <string[]>exclude)
```

**Review by christian-bromann — COMMENTED** [review]
Changeset looks good 👍 do you think we can add some smoke tests? We should already have some for applying keyword spec params.

Thanks a lot!

**Comment by tech-dm-klymenko** [comment]
@christian-bromann 
Sure! I will cover this functionality with smoke tests

File: packages/wdio-config/src/node/ConfigParser.ts (hunk 1 above)
Comment (tech-dm-klymenko) [reply packages/wdio-config/src/node/ConfigParser.ts:272]: Done

File: packages/wdio-config/src/node/ConfigParser.ts (hunk 3 above)
Comment (tech-dm-klymenko) [reply packages/wdio-config/src/node/ConfigParser.ts:320]: Done

File: packages/wdio-config/src/node/ConfigParser.ts (hunk 2 above)
Comment (tech-dm-klymenko) [reply packages/wdio-config/src/node/ConfigParser.ts:276]: Done

**Comment by tech-dm-klymenko** [comment]
@christian-bromann 
I have added smoke tests and updated documentation accordingly. 
Please review

File: packages/wdio-config/src/node/ConfigParser.ts [hunk 4]
```diff
@@ -479,4 +505,8 @@ export default class ConfigParser {
         const end = current === total ? undefined : specsPerShard * current
         return specs.slice(current * specsPerShard - specsPerShard, end)
     }
-}
\ No newline at end of file
+}
+
+function allKeywordsContainPath(excludedSpecList: string[]) {
+    return excludedSpecList.every(val => val.includes('/') || val.includes('\\'))
```
Comment (christian-bromann) [inline packages/wdio-config/src/node/ConfigParser.ts:511]:
```suggestion
    return excludedSpecList.every(val => val.includes(path.sep))
```

**Review by christian-bromann — APPROVED** [review]
One optimization but otherwise LGTM 👍

**Pull request merged**